### PR TITLE
Refactor/tidy up

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

started life as a tidy and ended up as a refactor to (hopefully) make things clearer and more testable...

- move code around to make it more testable
- isolate types to usage location
- add async capability to jest tests
- lints (and fixes) `import` order to keep diffs cleaner

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -45,6 +45,36 @@
...
+					"object",
+					"unknown"
+				],
+				"newlines-between": "never",
+				"alphabetize": {
+					"order": "asc" /* sort in ascending order. Options: ['ignore', 'asc', 'desc'] */,
+					"caseInsensitive": true /* ignore case. Options: [true, false] */
+				}
+			}
+		],
```
Comment (mxdvl) [inline .eslintrc.json:67]: That's convenient!

File: src/ccpa/types/CCPAConsentState.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export interface CCPAConsentState {
+	doNotSell: boolean;
+}
```
Comment (mxdvl) [inline src/ccpa/types/CCPAConsentState.ts:3]: Why use one file per type?

File: src/index.test.js [hunk 3]
```diff
@@ -31,7 +31,9 @@ beforeEach(() => {
 
 describe('cmp.init', () => {
 	it('requires isInUsa to be true or false', () => {
-		expect(cmp.init).toThrow();
+		expect(() => {
+			cmp.init({ pubData: {} });
+		}).toThrow('required');
```
Comment (mxdvl) [inline src/index.test.js:36]: What does `'required'` mean in this context?

**Review by mxdvl — APPROVED** [review]
Ended up as a big refactor: nice! 👍

File: src/ccpa/types/CCPAConsentState.ts (hunk 2 above)
Comment (sndrs) [reply src/ccpa/types/CCPAConsentState.ts]: moved all the types into `src/types`

**Comment by sndrs** [comment]
too big, breaking this down

**Pull request closed without merging**